### ISSUE TITLE
joinExisting aggregation barfs in ToolsUI with ncml namespace prefix elements

### ISSUE THREAD

**Issue by rsignell-usgs** [Unidata/thredds#447]
I just spend a super frustrating two hours figuring out a problem with
a joinExisting NcML aggregation.

I opened one file from the collection and used toolsUI to write the NcML.

I then added the aggregation element with appropriate regex.

When I tried to load this NcML aggregation back into ToolsUI, it
barfed, saying "no files found in aggregation".  After checking and
rechecking the regexp several times, I eventually figured out that
ToolsUI was barfing on the "ncml:" namespace specification on the
elements.   When I removed them it worked fine.

The irony here is that ToolsUI wrote those elements with the namespace
to begin with!

If you want to duplicate my problem, the two NcML files (one that
doesn't work with the namespace specification, and one without that
works) and two sample file s from the aggregation are here:

https://drive.google.com/open?id=0BzAHlPEEP_ujd25fdlRNZzJ1RjA

**Comment by dopplershift** [comment]
That's weird. I'm working on running it down, but it doesn't appear to be a 4.6.4 breakage. As a workaround if you need to do this more, using:
```sh
java -cp build/libs/toolsUI-4.6.5-SNAPSHOT.jar ucar.nc2.NCdumpW ~/rich_test/WHOI_ISLE_HFR_2014_07_01_800mgrid_1000mrad_20-Feb-2016.nc -ncml
```
gives NcML without the `ncml` namespace. I have no idea why those two aren't the same.

**Comment by rsignell-usgs** [comment]
You mean you wonder why clicking the NcML tab in ToolsUI GUI and the above command are not the same?

**Comment by dopplershift** [comment]
Correct.

This is fixed on 5.0--the ncml from both sources are the same, and do not have `ncml:` prefixed tags.

**Comment by dopplershift** [comment]
Oh, and 5.0 also fails to read in the one with the prefixes.

**Comment by rsignell-usgs** [comment]
okay, thanks for the clarification.

**Comment by dopplershift** [comment]
AHA! The `<scan>` element is missing the `ncml:` prefix:
```xml
<ncml:aggregation dimName="t" type="joinExisting">
    <scan location="." regExp=".*WHOI_ISLE_HFR_[0-9]{4}_[0-9]{2}_[0-9]{2}_800mgrid_1000mrad_20\-Feb\-2016\.nc$"/>
</ncml:aggregation>
```

**Comment by cwardgar** [comment]
I found the problem. At the bottom of `agg_with_namespace.ncml`:
````xml
<ncml:aggregation dimName="t" type="joinExisting">
    <scan location="." regExp=".*WHOI_ISLE_HFR_[0-9]{4}_[0-9]{2}_[0-9]{2}_800mgrid_1000mrad_20\-Feb\-2016\.nc$"/>
</ncml:aggregation>
````
The `scan` element needs an `ncml:` prefix. After adding that, ToolsUI reads it in okay. It's good to know that our NcmlReader properly handles the `ncml` namespace when it's default and when it's explicitly prefixed to each element.

But yeah, I think there are 3 different classes that write NcML in 4.6, and some of them produce incorrect output, as you've seen. In 5.0, I rewrote all that. All NcML writing is done by a single class, using a default namespace.
